### PR TITLE
SignaturePad: fix scaling on smaller screens

### LINKED ISSUE #5132: [Bug]: SignaturePad: Repeating Images When Used in Mobile Device Dimensions
### Blazorise Version

1.3.2

### What Blazorise provider are you running on?

Bootstrap5

### Link to minimal reproduction, or a simple code snippet

[https://bootstrap5demo.blazorise.com/tests/signaturepad](https://bootstrap5demo.blazorise.com/tests/signaturepad)

Can repro this behavior from the Demo site (https://bootstrap5demo.blazorise.com/tests/signaturepad) on mobile devices (iOS and Andoid using Chrome and Edge) and also in desktop Chrome with the developer tools panel open and the device toolbar set to any mobile device size (I used iPad Mini for most tests).

This is a duplicate of the closed issue #4841 

image

### Steps to reproduce

1. Using Chrome, navigate to [https://bootstrap5demo.blazorise.com/tests/signaturepad](https://bootstrap5demo.blazorise.com/tests/signaturepad)
2. Open Dev Tools panel
3. Toggle the Device Toolbar on (Ctrl + Shift + M)
4. Set the Dimensions to iPad Mini (or any mobile device in the list)
5. Draw on SignaturePad (both mouse and touchscreen have the same behavior)
6. Note the signature image is repeated (reflected?) above the drawen signature
7. Alos note, if the resulting signature image is saved (not in the demo site of course), the saved image also contains the repeated signatures

### What is expected?

A single signature as drawn on both the SignaturePad and in the resulting image when saved.

### What is actually happening?

As shown in the screenshot, the drawn signature is repeated several times in the SignaturePad.

### What browsers are you seeing the problem on?

Chrome, Microsoft Edge

### Any additional comments?

_No response_

### PR DESCRIPTION
**Pull request by stsrki**
Closes #5132

### CODE REVIEW

**Comment by stsrki** [comment]
@David-Moreira reminder

File: Source/Extensions/Blazorise.SignaturePad/wwwroot/signaturepad.js [hunk 1]
```diff
@@ -176,4 +185,42 @@ function getImageDataURL(sigpad, options) {
...
+
+function resizeCanvas(sigpad, canvas) {
+    if (!sigpad) {
+        return;
+    }
+
+    if (!canvas) {
+        return;
+    }
+
```
Comment (David-Moreira) [inline Source/Extensions/Blazorise.SignaturePad/wwwroot/signaturepad.js:212]: If you want to bother inlining in the same if

File: Source/Extensions/Blazorise.SignaturePad/wwwroot/signaturepad.js [hunk 2]
```diff
@@ -176,4 +185,42 @@ function getImageDataURL(sigpad, options) {
...
+    const ratio = Math.max(window.devicePixelRatio || 1, 1);
+
+    const context = canvas.getContext("2d");
+
+    // This part causes the canvas to be cleared
+    canvas.width = canvas.offsetWidth * ratio;
+    canvas.height = canvas.offsetHeight * ratio;
+    context.scale(ratio, ratio);
+
+    sigpad.fromData(sigpad.toData());
```
Comment (David-Moreira) [inline Source/Extensions/Blazorise.SignaturePad/wwwroot/signaturepad.js:225]: I guess this is a way to refresh? Do they not have a refresh or reset or whatever method? So you don't have to incur the costs of `toData` and `FromData`? Altought they might be negligible, I don't know.

**Review by David-Moreira — APPROVED** [review]
LGTM, as long as it works :D

File: Source/Extensions/Blazorise.SignaturePad/wwwroot/signaturepad.js (hunk 2 above)
Comment (stsrki) [reply Source/Extensions/Blazorise.SignaturePad/wwwroot/signaturepad.js:225]: This is per their instructions so as far as I know there is nothing built in that does the same.

**Pull request merged**